ecs-cli/modules/utils/compose/ecs_params_reader.go

```diff
@@ -61,3 +83,39 @@ func readECSParams(filename string) (*ECSParams, error) {
 
 	return ecsParams, nil
 }
+
+// ConvertToEcsNetworkConfiguration extracts out the NetworkConfiguration from
+// the ECSParams into a format that is compatible with ECSClient calls.
+func ConvertToEcsNetworkConfiguration(ecsParams *ECSParams) (*ecs.NetworkConfiguration, error) {
+	networkConfig := ecsParams.RunParams.NetworkConfiguration
+	awsvpcConfig := networkConfig.AwsVpcConfiguration
+	subnets := awsvpcConfig.Subnets
+	securityGroups := awsvpcConfig.SecurityGroups
+	networkMode := ecsParams.TaskDefinition.NetworkMode
+
+	if len(subnets) < 1 && networkMode == "awsvpc" {
```

Comment:
minor: if you flip the order, this code would be more efficient.

Reply:
ah, nice catch. fixed.